Do not use `--package geneweb` to build GeneWeb

Halbaroth opened the pull request:
The target `build` of the Makefile runs the command:
```sh
dune build -p geneweb
```
As the man of `dune build` explains, the option `--package` is equivalent to `--only-package` AND `--release`. In the release profile, the warnings are not fatals and it seems that some of them are not printed at all!

Nowadays, compiling is quick and there is no reason to limit the scope of `make build` to the `geneweb` package.

This PR changes the Makefile as follows:
1. `make build` builds everything.
2. Remove `gwd` as `build` builds everything.

Halbaroth commented:
The CI fails because #2147 is not merged and ocaml-setup v2 is dead.

a2line commented:
Why do you remove `make gwd` that builds only gwd/gwc exe? I added it last year and use it every day in local (old slow computer) and also on the Colab script where it shorten builds a lot installing less deps + less to compile (only the two exe needed to create a database and launch gwd).

Halbaroth committed:
The target `build` of the Makefile run the command:
```sh
dune build -p geneweb
```
As the man of `dune build` explains, the option `--package` is
equivalent to `--only-package` AND `--release`. In the release profile,
the warnings are not fatals and it seems that some of them are not
printed at all!

Nowadays, compiling is quick and there is no reason to limit the scope
of `make build` to the `geneweb` package.
